### DISCUSSION TITLE
open api to get storage_logs

### DISCUSSION THREAD

**Discussion by UrsaMajor-t** [zkSync-Community-Hub/zksync-developers#19] [Feature requests]
### Network

zkSync Era

### Provide a brief description of the feature you would like to see implemented.

It is hoped that an api can be opened to obtain storage_logs, which is of great help to the third-party implementation of zksync era's certification system.

### Who does this feature request benefit the most (users, developers, protocol, etc.)?

developers and protocol

**Comment by uF4No** [comment]
Hi @LAMOUR0913 

Can you elaborate a little bit more and explain some use cases? 
What would you expect to retrieve from this API? If a slot has already been updated in a batch so it'll be free for you to update it again?

**Comment by UrsaMajor-t** [comment]
@uF4No If the stored logs are available, I can calculate the roothash corresponding to the l1 batch from a batch of tx stored logs. (The roothash of batch l1 is known to be the Merkle tree roothash of all tx storage logs of the current batch.)
This is useful for developers or the protocols that support your network environment to prove the existence of a transaction

**Comment by dutterbutter** [comment]
Thank you for reaching out with your feature request. We appreciate the time you've taken to provide valuable input on how we can improve our system.

The idea to expose an API for obtaining storage_logs is a compelling one that we will be evaluating! 🚀

**Reply by UrsaMajor-t** [reply]
Is there any information at present? @dutterbutter
